### PR TITLE
[GPU] Tile blocked layouts support

### PR DESCRIPTION
**Pull request by ozhydkov-lohika**
Added Tile blocked layouts support

### Tickets:
 - 85190

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/ocl/tile.cpp [hunk 1]
```diff
@@ -46,23 +46,31 @@ struct tile_impl : typed_primitive_impl_ocl<tile> {
...
+            format::bs_fs_zyx_bsv32_fsv16
+    };
+
+    std::set<std::tuple<data_types, format::type>> keys;
+    for (const auto& t : types) {
+        for (const auto& f : formats) {
+            keys.emplace(t, f);
+        }
+    }
+    implementation_map<tile>::add(impl_types::ocl, tile_impl::create, keys);
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/src/graph/impls/ocl/tile.cpp:73]: we already have overloaded method to have two params formats and types, so we don't need to manually introduce "keys" here

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/tile/tile_kernel_ref.cpp [hunk 2]
```diff
@@ -26,6 +26,30 @@ ParamsKey TileKernelRef::GetSupportedKey() const {
...
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv16_fsv32);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv16_fsv32);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv16_fsv16);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv16_fsv16);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv16_fsv16);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv16_fsv16);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv32_fsv32);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv32_fsv32);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv32_fsv16);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv32_fsv16);
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/tile/tile_kernel_ref.cpp:52]: as here is almost all the layouts already I think it's acceptable to change to
EnableAllIntputLayouts and EnableAllOutputLayouts here

File: src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp [hunk 3]
```diff
@@ -243,3 +243,427 @@ TEST(tile_gpu, basic_in1x2x2x2_axis_z) {
...
+                    getValues<T>({
+                                   1.f, 1.f, 1.f, 1.f,
+
+                                   0.f, 0.f, 0.f, 0.f,
+
+                                   5.f, 5.f, 5.f, 5.f,
+
+                                   1.5f, 1.5f, 1.5f, 1.5f,
+                                 }),
+                    tile::along_x,
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp:390]: have you used clang-format for this file?

File: src/plugins/intel_gpu/src/graph/impls/ocl/tile.cpp (hunk 1 above)
Comment (ozhydkov-lohika) [reply src/plugins/intel_gpu/src/graph/impls/ocl/tile.cpp]: Done

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/tile/tile_kernel_ref.cpp (hunk 2 above)
Comment (ozhydkov-lohika) [reply src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/tile/tile_kernel_ref.cpp]: Done

File: src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp (hunk 3 above)
Comment (ozhydkov-lohika) [reply src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp]: Done

**Comment by wilson-seok** [comment]
@ozhydkov-lohika Please resolve PR commits/check failure. I think you can access to error log. If not, please let me know.

**Comment by wilson-seok** [comment]
No performance regression during int8 models.

**Comment by yeonbok** [comment]
@ozhydkov-lohika the tile operation is not added to the can_use_fsv16 flag's white list in program.cpp. 
I guess that the network is not mapped to fsv16.
@wilson-seok  Sorry to say that but, could you add it to whitelist and check performance again :( ?

**Comment by tgubanova-lohika** [comment]
> @ozhydkov-lohika the tile operation is not added to the can_use_fsv16 flag's white list in program.cpp. I guess that the network is not mapped to fsv16. @wilson-seok Sorry to say that but, could you add it to whitelist and check performance again :( ?

It's funny, but I've investigated that sometimes blocked layouts work (at least in unit tests) for some primitives even without mentioning them in program.cpp. But still don't clearly understand the dependency

**Comment by yeonbok** [comment]
@tgubanova-lohika In unittest, wasnt it enforced the layout? Then it will use the blocked format. However when we run the cldnn from the gpu plugin (inference engine), it is not done so. First the program is checkign whether the network "can use blocked format " using that whitelist + other conditions (e.g., # of convolutions etc..) and then it decides the prefered format layer by layer at reorder_inputs.

**Comment by wilson-seok** [comment]
No performance regression during int8 models today.

**Comment by wilson-seok** [comment]
Please fix build error in build_rhel82_release.
image

**Comment by wilson-seok** [comment]
Please fix build error in build_ubuntu18_release.
image

**Review by wilson-seok — APPROVED** [review]

File: src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp [hunk 4]
```diff
@@ -249,3 +249,438 @@ TEST(tile_gpu, basic_in1x2x2x2_axis_z) {
...
+namespace {
+template<typename T>
+struct Params {
+    tensor input_tensor;
+    std::vector<T> inputs;
+    std::vector<int64_t> repeats;
+    tensor output_tensor;
+    std::vector<T> outputs;
+    int output_axis;
+    int num_tiles;
```
Comment (kelvinchoi-intel) [inline src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp:262]: Where are output_axis and num_tiles used?

**Comment by kelvinchoi-intel** [comment]
LGTM

**Comment by wilson-seok** [comment]
@ozhydkov-lohika Please resolve conflicts.

File: src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp (hunk 4 above)
Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/tests/test_cases/tile_gpu_test.cpp]: done, removed

**Review by kelvinchoi-intel — APPROVED** [review]

**Pull request merged**